### ISSUE TITLE
Warning about empty needle in stristr() polluting log file

### ISSUE THREAD

**Issue by webmaster-cses-org-uk** [TechToThePeople/publicautocomplete#43]
In the `getpublic` API method definition, there is no protection against passing an empty `$term` to `stristr()` which causes warnings of the following nature to appear in the log file every time the autocomplete field is cleared (text deleted):

```
[Sun Jul 03 21:34:13.208174 2022] [fcgid:warn] [pid 124034] [client 2.100.74.70:55269] mod_fcgid: stderr: PHP Warning:  stristr(): Empty needle in /home/cses_org_uk/public_html/media/civicrm/ext/eu.tttp.publicautocomplete/api/v3/Contact/Getpublic.php on line 80, referer: https://cses.org.uk/education/bursary/apply
```

A simple guard to first make sure that `$term` is not empty before calling `stristr()` fixes this. PR to follow.

**Comment by webmaster-cses-org-uk** [comment]
PR #44 implements proposed fix.

**Comment by twomice** [comment]
Thanks @webmaster-cses-org-uk I'll be looking at this next Monday.

**Comment by twomice** [comment]
Fixed in 227bfcee5d22b581d1c2ab64f72e7e6ae843851a.
